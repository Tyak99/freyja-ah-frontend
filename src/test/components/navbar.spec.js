import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navbar from '../../components/navbar/navbar';

const setUp = (props = {}) => {
  const component = shallow(<Navbar {...props} />);
  return component;
};
describe('Navbar component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it('should render without errors', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should render children', () => {
    const wrapper = component.find('Navbar');
    expect(wrapper.find('NavbarBrand').exists()).toBe(true);
    expect(wrapper.find('Collapse').exists()).toBe(true);
    expect(wrapper.find('NavbarToggler').exists()).toBe(true);
    expect(wrapper.find('Nav').exists()).toBe(true);
    expect(wrapper.find('NavItem').exists()).toBe(true);
    expect(wrapper.find('NavLink').exists()).toBe(true);
    expect(wrapper.find('.logo').exists()).toBe(true);
  });
  it('should simulate click', () => {
    const navbar = component.find('Navbar');
    expect(navbar.find('NavbarToggler').simulate('click'));
  });
});
