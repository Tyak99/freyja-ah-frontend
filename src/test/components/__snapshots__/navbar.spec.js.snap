// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render without errors 1`] = `
<div>
  <Navbar
    expand="md"
    light={true}
    tag="nav"
  >
    <NavbarBrand
      href="/"
      tag="a"
    >
      <img
        alt="Logo"
        className="logo"
        src=""
      />
    </NavbarBrand>
    <NavbarToggler
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <NavLink
            className="navc"
            href="/"
            tag="a"
          >
            Home
          </NavLink>
        </NavItem>
        <NavItem
          tag="li"
        >
          <NavLink
            className="navc"
            href="/login"
            tag="a"
          >
            Sign In
          </NavLink>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;
